src/display/annotation_layer.js

```diff
@@ -305,127 +306,35 @@ var TextAnnotationElement = (function TextAnnotationElementClosure() {
      * @returns {HTMLSectionElement}
      */
     render: function TextAnnotationElement_render() {
-      var rect = this.data.rect, container = this.container;
-
-      // Sanity check because of OOo-generated PDFs.
```

Comment:
Right now this code appears to have no effect because the container is already created before we get here, meaning that the size of the container is already fixed, no matter how we change the rectangle here.

The only line that used these changed values was https://github.com/mozilla/pdf.js/pull/6792/files#diff-1097e3b131ba22f8e8a6c2f20e3ab5ecL332. It was used to position the wrapper with the popup next to the Text annotation icon with a hardcoded margin of 5 pixels. In the new code this is achieved by https://github.com/mozilla/pdf.js/pull/6792/files#diff-1097e3b131ba22f8e8a6c2f20e3ab5ecR331 and the margin in CSS at https://github.com/mozilla/pdf.js/pull/6792/files#diff-7d2639c28c78de0283c24b2c5d56eb60R55.

If for some reason the rectangle of an annotation cannot be trusted (which I have yet to see in practice and I could not find a test case for which this code has been added), then this is also the wrong place because it would apply to all annotations then. In that case the PDF would be incorrect and the specification does not describe what to do then, so displaying a small annotation at the size of the rectangle should also be fine. I did notice that Adobe Reader can display very small annotations that have been created like that on purpose (for example a small dot), so respecting the original rectangle and not enforcing a minimal size seems more correct to me too.

Would you agree or did I miss something in my reasoning?